Lib9c/Action/CustomEquipmentCraft/CustomEquipmentCraft.cs

```diff
@@ -82,197 +99,190 @@ public override IWorld Execute(IActionContext context)
+                    if (!avatarState.inventory.RemoveMaterial(itemId, context.BlockIndex, amount))
                     {
                         throw new NotEnoughItemException(
-                            $"Insufficient material {materialCost.ItemId}: {materialCost.Amount} needed"
+                            $"Insufficient material {itemId}: {amount} needed"
                         );
                     }
-
-                    requiredFungibleItems[materialCost.ItemId] = materialCost.Amount;
                 }
-            }
 
-            // Select data to create equipment
-            var random = context.GetRandom();
-            var endBlockIndex = context.BlockIndex +
-                                (long)Math.Floor(recipeRow.RequiredBlock *
-                                                 relationshipRow.RequiredBlockMultiplier);
+                var endBlockIndex = context.BlockIndex + (long)Math.Floor(
+                    recipeRow.RequiredBlock * relationshipRow.RequiredBlockMultiplier / 10000m
+                );
 
-            var iconId = ItemFactory.SelectIconId(
-                IconId, IconId == RandomIconId, relationship,
-                sheets.GetSheet<CustomEquipmentCraftIconSheet>(), random
-            );
-            var optionRow = ItemFactory.SelectOption(
-                recipeRow.ItemSubType, sheets.GetSheet<CustomEquipmentCraftOptionSheet>(), random
-            );
-            var skill = ItemFactory.SelectSkill(
-                recipeRow.ItemSubType,
-                sheets.GetSheet<CustomEquipmentCraftRecipeSkillSheet>(),
-                sheets.GetSheet<EquipmentItemOptionSheet>(),
-                sheets.GetSheet<SkillSheet>(),
-                random
-            );
+                // Create equipment with ItemFactory
+                var guid = random.GenerateRandomGuid();
+                var equipment =
+                    (Equipment)ItemFactory.CreateItemUsable(equipmentRow, guid, endBlockIndex);
 
-            // Create equipment with ItemFactory
-            var equipment = ItemFactory.CreateCustomEquipment(
-                random,
-                iconId,
-                equipmentRow,
-                endBlockIndex,
-                avatarState.level,
-                relationshipRow,
-                optionRow,
-                skill
-            );
+                // Set Icon
+                equipment.IconId = ItemFactory.SelectIconId(
+                    craftData.IconId, craftData.IconId == RandomIconId, equipmentRow, relationship,
+                    sheets.GetSheet<CustomEquipmentCraftIconSheet>(), random
+                );
 
-            // Add equipment
-            avatarState.inventory.AddItem(equipment);
-            avatarState.blockIndex = context.BlockIndex;
-            avatarState.updatedAt = context.BlockIndex;
+                // Set Elemental Type
+                var elementalList = Enum.GetValues<ElementalType>();
```

Comment:
```suggestion
                var elementalList = Enum.GetValues(typeof(ElementalType));
```
유니티에서는 `Enum.GetValues<T>()` 를 지원하지 않아서 (아마 닷넷 버전 차이인듯) typeof를 쓰는 형태로 바꿔주셔야 유니티에서도 컴파일이 됩니다.